### DISCUSSION TITLE
How to use "live grep" with arguments ?

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#2537] [General]
Example; If want to search a word under a specific **file type** like **java**, I would do: `-tjava` in ripgrep.

How do I accomplish this using live grep here?

**Comment by dpetka2001** [comment]
#2292 should be of help. You need to install another plugin `telescope-live-grep-args`.

**Reply by idelice** [reply]
I didn't thought any additional plugin was needed but ok. This works.

FYI or anyone else reading this and can't make the mapping configuration work, then the reply here fixed it for me:
https://github.com/LazyVim/LazyVim/discussions/1918#discussioncomment-7448227 

Basically, instead of adding the configuration for the live grep arg plugin inside where I add it as a plugin, I add the mapping configuration inside the telescope's opts like so:

```lua
    opts = function(_, opts)

    local lga_actions = require("telescope-live-grep-args.actions")

    opts.extensions = {
      live_grep_args = {
        auto_quoting = true, -- enable/disable auto-quoting
        -- define mappings, e.g.
        mappings = { -- extend mappings
          i = {
            ["<C-k>"] = lga_actions.quote_prompt(),
            ["<C-t>"] = lga_actions.quote_prompt({ postfix = " -g \"*Test*\"" }),
            ["<C-j>"] = lga_actions.quote_prompt({ postfix = " -g \"!*Test*\"" }),
          },
        },
        -- ... also accepts theme settings, for example:
        -- theme = "dropdown", -- use dropdown theme
        -- theme = { }, -- use own theme spec
        -- layout_config = { mirror=true }, -- mirror preview pane
        }
        end
```
